File: rb/features.rb
# ViacepAddressLookup SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module ViacepAddressLookupFeatures
  def self.make_feature(name)
    case name
    when "base"
      ViacepAddressLookupBaseFeature.new
    when "test"
      ViacepAddressLookupTestFeature.new
    else
      ViacepAddressLookupBaseFeature.new
    end
  end
end

File: py/features.py
# ViacepAddressLookup SDK feature factory

from feature.base_feature import ViacepAddressLookupBaseFeature
from feature.test_feature import ViacepAddressLookupTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ViacepAddressLookupBaseFeature(),
        "test": lambda: ViacepAddressLookupTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
